tasksCode/AnyType.hpp

```diff
@@ -0,0 +1,412 @@
+	// generally always throw bad_cast exception
+	template <typename Type>
+    explicit operator Type() const {
+		throw std::bad_cast{ };
+	};
+
+	// Return value' type
+	std::string GetValueType() const;
+	
+	// Delete internal data
+	void DeleteData();
+	
+	// Swap two objects
+	void Swap(AnyType & anyType);
+	
+	// Swap two objects
+	friend void Swap(AnyType & anyTypeA, AnyType & anyTypeB);
+
+	// Default destructor
+	~AnyType() {
+		delete valueData;
+	}
+};
+
+// Set the default internal value' data
+void AnyType::SetValueData()
+{
+	if (valueData == nullptr)
+		valueData = new ValueData;
+	// Set all valueData bytes to 0 (the long long int has 64 bytes
+	// which is the max fundamental datatype size)
+	(*valueData).valueLLInt = 0;
+}
+
+// Set the internal value' data, based on the give value
+template <typename Type>
+void AnyType::SetValueData(const Type & value)
+{
+	if (valueData == nullptr)
+		valueData = new ValueData;
+	// Iterate through possible types to define what value interpretation should be choosen
+	// The write the value into the value' data according to the correct interpreation
+	auto valueTypeId = typeid(value).name();
+	if		(valueTypeId == typeid( bool					).name()) (*valueData).valueBool = static_cast<bool>(value);
+	else if (valueTypeId == typeid( char					).name()) (*valueData).valueChar = static_cast<char>(value);
+	else if (valueTypeId == typeid( unsigned char			).name()) (*valueData).valueUChar = static_cast<unsigned char>(value);
+	else if (valueTypeId == typeid( int						).name()) (*valueData).valueLLInt = static_cast<long long>(value);
+	else if (valueTypeId == typeid( short int				).name()) (*valueData).valueLLInt = static_cast<long long>(value);
+	else if (valueTypeId == typeid( long int				).name()) (*valueData).valueLLInt = static_cast<long long>(value);
+	else if (valueTypeId == typeid( long long int			).name()) (*valueData).valueLLInt = static_cast<long long>(value);
+	else if (valueTypeId == typeid( unsigned int			).name()) (*valueData).valueLLUInt = static_cast<unsigned long long>(value);
+	else if (valueTypeId == typeid( unsigned short int		).name()) (*valueData).valueLLUInt = static_cast<unsigned long long>(value);
+	else if (valueTypeId == typeid( unsigned long int		).name()) (*valueData).valueLLUInt = static_cast<unsigned long long>(value);
+	else if (valueTypeId == typeid( unsigned long long int	).name()) (*valueData).valueLLUInt = static_cast<unsigned long long>(value);
+	else if (valueTypeId == typeid( float					).name()) (*valueData).valueLDouble = static_cast<long double>(value);
+	else if (valueTypeId == typeid( double					).name()) (*valueData).valueLDouble = static_cast<long double>(value);
+	else if (valueTypeId == typeid( long double				).name()) (*valueData).valueLDouble = static_cast<long double>(value);
+	else throw std::bad_typeid{ }; // Cast exception if the type is unknown
+}
+
+// Set the default internal value' type
+void AnyType::SetValueType()
```

Comment:
I'd suggest to name this fucntion using "void". FE: SetValueTypeVoid or stm like that

Reply:
This is default value type setter. Of course since there is no value it sets its value to void. See template parametrized value setter.